pilot/pkg/bootstrap/server.go

```diff
@@ -232,6 +232,12 @@ func NewServer(args *PilotArgs, initFuncs ...func(*Server)) (*Server, error) {
 	// used for both initKubeRegistry and initClusterRegistries
 	args.RegistryOptions.KubeOptions.EndpointMode = kubecontroller.DetectEndpointMode(s.kubeClient)
 
+	enableEndpointSlice, _ := features.EnableEndpointSliceController()
```

Comment:
Do you still need this when EndpointSlice is used by default?

Reply:
So EndpointSlices are only turned on for K8s 1.21+. This acts as a way to also help us feature-gate dual stack from being run on earlier versions, although we should have a check similar to what was done for https://github.com/istio/istio/pull/34971 instead

Reply:
Yeah, I think we need to consider a situation that there is a custom set to false, then the mistake can be caught by following logic.